### ISSUE TITLE
Design Time Host on Windows not Starting

### ISSUE THREAD

**Issue by brad-jones** [OmniSharp/omnisharp-atom#450]
This used to work fine, not sure if it's my system that has changed or something in omnisharp-atom.
But what ever it is that has changed, has happen is the last few days.

Output from Atom:
```
Starting OmniSharp server (pid:7028)
OmniSharp Location: C:\Users\Brad\.atom\packages\omnisharp-atom\node_modules\omnisharp-client\roslyn\omnisharp.cmd
Change the location that OmniSharp is loaded from by setting the OMNISHARP environment variable
OmniSharp Path: C:\Users\Brad\Documents\Projects\dotneteloquent
Omnisharp server running using stdio at location 'C:\Users\Brad\Documents\Projects\dotneteloquent' on host 7028.
Looking for sdk version in 'C:\Users\Brad\Documents\Projects\dotneteloquent\global.json'.
Using runtime 'C:\Users\Brad\.dnx\runtimes\dnx-clr-win-x64.1.0.0-beta6-12208'.
Scanning 'C:\Users\Brad\Documents\Projects\dotneteloquent' for DNX projects
Found project 'C:\Users\Brad\Documents\Projects\dotneteloquent\tests\DotNetEloquent.Tests\project.json'.
Found project 'C:\Users\Brad\Documents\Projects\dotneteloquent\src\DotNetEloquent\project.json'.
Running DesignTimeHost on port 1923, with PID 3480

The project system 'DnxProjectSystem' threw an exception.
System.Net.Sockets.SocketException (0x80004005): No connection could be made because the target machine actively refused it 127.0.0.1:1923
   at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)
   at OmniSharp.Dnx.DnxProjectSystem.<>c__DisplayClass13_4.b__0(Int32 port)
   at OmniSharp.Dnx.DesignTimeHostManager.Start(String hostId, Action`1 onConnected)
   at OmniSharp.Dnx.DnxProjectSystem.Initalize()
   at OmniSharp.Startup.Configure(IApplicationBuilder app, ILoggerFactory loggerFactory, IOmnisharpEnvironment env, ISharedTextWriter writer)

No solution files found in 'C:\Users\Brad\Documents\Projects\dotneteloquent'
Detecting CSX files in 'C:\Users\Brad\Documents\Projects\dotneteloquent'.
Solution has finished loading
Could not find any CSX files
started
Listening on port 1923
/projects: 200 579ms
/codecheck: 200 562ms
/codecheck: 200 562ms
/codecheck: 200 562ms
```

If I start omnisharp directly via cmd prompt I get:

```
info    : [OmniSharp.Startup] Omnisharp server running on port '2000' at location 'C:\Users\Brad\Documents\Projects\dotneteloquent' on host -1.
info    : [OmniSharp.Dnx.DnxPaths] Looking for sdk version in 'C:\Users\Brad\Documents\Projects\dotneteloquent\global.json'.
info    : [OmniSharp.Dnx.DnxPaths] Using runtime 'C:\Users\Brad\.dnx\runtimes\dnx-clr-win-x64.1.0.0-beta6-12208'.
info    : [OmniSharp.Dnx.DnxProjectSystem] Scanning 'C:\Users\Brad\Documents\Projects\dotneteloquent' for DNX projects
info    : [OmniSharp.Dnx.DnxProjectSystem] Found project 'C:\Users\Brad\Documents\Projects\dotneteloquent\tests\DotNetEloquent.Tests\project.json'.
info    : [OmniSharp.Dnx.DnxProjectSystem] Found project 'C:\Users\Brad\Documents\Projects\dotneteloquent\src\DotNetEloquent\project.json'.
info    : [OmniSharp.Dnx.DesignTimeHostManager] Running DesignTimeHost on port 1872, with PID 4368
info    : [OmniSharp.Dnx.DnxProjectSystem] Connected
info    : [OmniSharp.MSBuild.MSBuildProjectSystem] No solution files found in 'C:\Users\Brad\Documents\Projects\dotneteloquent'
info    : [OmniSharp.ScriptCs.ScriptCsProjectSystem] Detecting CSX files in 'C:\Users\Brad\Documents\Projects\dotneteloquent'.
info    : [OmniSharp.ScriptCs.ScriptCsProjectSystem] Could not find any CSX files
info    : [OmniSharp.Startup] Solution has finished loading
```

I will note that a Windows Firewall exception popped up when I started it from the console.
I now have the Windows Firewall completely disabled and it still does not work from Atom.

I have also tried running Atom as Administrator, thought it might have been something to do with elevated privileges but no change.

Seeing as I can start it from console manually. Is there a quick hack I can do to tell atom not start it's own instance of omnisharp server and just use the one I started???

Anything else I should check?

Cheers Brad

**Comment by brad-jones** [comment]
Oh forgot to mention, using the latest version: v0.22.2

**Comment by brad-jones** [comment]
Okay I have narrowed it down, it's something to do with the global.json file.

Contents of which are:
```json
{
    "sources": ["src", "test"]
}
```

I setup my project as suggested here:
https://github.com/aspnet/Testing/wiki/How-to-create-test-projects

I have another project that just has a single project.json file and that works fine. The design time host starts and atom connects to it as it always has.

Thinking out load: Is it something like 2 process are trying to start on the same port when multiple projects are involved?

**Comment by brad-jones** [comment]
Okay found out that "sources" has been changed to "projects".
https://github.com/aspnet/Announcements/issues/18

However I don't know that this was the cause, it seems like there is a race condition somewhere.
As I have had Atom start omnisharp correctly a few times in the last hour or so.

**Comment by brad-jones** [comment]
Okay slowly getting my head around the inner workings of omnisharp.
This is obviously where the exception is being thrown:
https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp/Dnx/DesignTimeHostManager.cs

**Comment by brad-jones** [comment]
Sorry the exception is probably coming from: _designTimeHostManager.Start
https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp/Dnx/DnxProjectSystem.cs

**Comment by brad-jones** [comment]
If I run from console:

```
C:\Users\Brad\.atom\packages\omnisharp-atom\node_modules\omnisharp-client\roslyn\omnisharp.cmd --stdio
```

Then I get the same SocketException.

**Comment by brad-jones** [comment]
This is clearly something to do with omnisharp-roslyn, not the atom plugin it's self.

**Comment by nosami** [comment]
Nicely done! 

For reference, [this](https://github.com/OmniSharp/omnisharp-atom/wiki#modifying-the-server) is how you connect Atom to a custom server - although unusually, there is a breaking change in omnisharp-roslyn master right now that needs fixing on the Atom side.

**Comment by nosami** [comment]
@brad-jones your omnisharp-roslyn fix is in omnisharp-atom v0.22.4

**Comment by brad-jones** [comment]
Sweet, love the fast release cycle.
